Lib/urllib/parse.py

```diff
@@ -168,6 +168,8 @@ def port(self):
         port = self._hostinfo[1]
         if port is not None:
             try:
+                if not port.isdigit():
+                    raise ValueError(f"Port {port!r} contains non-numeric character(s)")
```

Comment:
This should be outside the `try`, because otherwise we'll wrap this ValueError into another ValueError below.